### PR TITLE
Add strip option for PTQ

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

1. API for strip model:
    - `nncf.strip(model, do_copy)`
    - `nncf.torch.strip(model, do_copy)`
    - `pt_model.nncf.strip(do_copy)` 

2. Removed conversion to torch native FQ for num_bits=4 (not exported by ONNX)
3. Fix warning of using old style nncf api

### Related tickets

106014, 107781

### Tests

test_nncf_strip_api
test_create_nncf_config

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
run openvino pre-commit tests

File: tests/tensorflow/quantization/test_strip.py [hunk 1]
```diff
@@ -52,3 +56,30 @@ def test_do_copy(do_copy):
...
+def test_nncf_quantize_strip(strip_model):
+    model = get_basic_two_conv_test_model()
+
+    def transform_fn(data_item):
+        return data_item[0]
+
+    dataset = nncf.Dataset(get_const_target_mock_regression_dataset(img_size=4), transform_fn)
+
+    if strip_model is not None:
+        advanced_parameters = AdvancedQuantizationParameters()
```
Comment (AlexKoff88) [inline tests/tensorflow/quantization/test_strip.py:71]: Can I do it this way? 
```py
advanced_parameters = AdvancedQuantizationParameters(strip_model=True)
```

Comment (AlexanderDokuchaev) [reply tests/tensorflow/quantization/test_strip.py]: Yes, it`s equivalently. Update in tests.

**Comment by AlexanderDokuchaev** [comment]
@AlexKoff88  for torch (not torch_ptq) on current state it's possible way. I will add example in next PR.

**Review by alexsu52 — COMMENTED** [review]
I initiated discussing this PR offline. Please do not merge until the issues are resolved.

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

File: nncf/experimental/torch/quantization/quantize_model.py [hunk 2]
```diff
@@ -103,6 +103,9 @@ def quantize_impl(
     if target_device == TargetDevice.CPU_SPR:
         raise RuntimeError("target_device == CPU_SPR is not supported")
 
+    if advanced_parameters is None:
```
Comment (alexsu52) [inline nncf/experimental/torch/quantization/quantize_model.py:106]: It looks like is not needed anymore.

File: nncf/quantization/advanced_parameters.py [hunk 3]
```diff
@@ -283,46 +284,52 @@ def convert_range_estimator_parameters_to_dict(params: RangeEstimatorParameters)
...
+    ):
+        return {}
     else:
         raise RuntimeError("The following range estimator parameters are not supported: " f"{str(params)}")
 
     return result
 
 
-def convert_advanced_parameters_to_dict(params: AdvancedQuantizationParameters) -> Dict[str, Any]:
+def apply_advanced_parameters_to_config(config: NNCFConfig, params: AdvancedQuantizationParameters) -> NNCFConfig:
```
Comment (alexsu52) [inline nncf/quantization/advanced_parameters.py:300]: `NNCFConfig` will be deprecated in the future. Could you prove the necessity of these changes?

**Review by alexsu52 — COMMENTED** [review]
Thanks for the contribution!

File: nncf/quantization/advanced_parameters.py (hunk 3 above)
Comment (AlexanderDokuchaev) [reply nncf/quantization/advanced_parameters.py]: Removed from this PR, https://github.com/openvinotoolkit/nncf/pull/1906

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

**Comment by AlexanderDokuchaev** [comment]
run tensorflow pre-commit tests

**Comment by AlexanderDokuchaev** [comment]
run tensorflow pre-commit tests

**Review by alexsu52 — APPROVED** [review]
LGTM

File: tests/torch/quantization/test_strip.py [hunk 4]
```diff
@@ -314,3 +312,21 @@ def test_do_copy(do_copy):
...
+    config = _get_config_for_algo(model.INPUT_SIZE)
+
+    quantized_model, _ = create_compressed_model_and_algo_for_test(model, config)
+
+    if strip_type == "nncf":
+        strip_model = nncf.strip(quantized_model)
+    elif strip_type == "torch":
+        strip_model = nncf.torch.strip(quantized_model)
+    elif strip_type == "nncf_interfere":
+        strip_model = quantized_model.nncf.strip()
```
Comment (AlexKoff88) [inline tests/torch/quantization/test_strip.py:329]: Is this still a valid way `quantized_model.nncf.strip()`?

**Review by AlexKoff88 — APPROVED** [review]

File: nncf/torch/nncf_network.py [hunk 5]
```diff
@@ -738,6 +738,20 @@ def get_node_to_op_address_mapping(self) -> Dict[NNCFNodeName, OperationAddress]
     def set_compression_controller(self, ctrl: "PTCompressionAlgorithmController"):
         self.compression_controller = ctrl
 
+    def strip(self, do_copy: bool = True) -> "NNCFNetwork":
+        """
+        _summary_
```
Comment (vshampor) [inline nncf/torch/nncf_network.py:743]: What is this?

**Review by vshampor — CHANGES_REQUESTED** [review]

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

**Pull request merged**